ordersettler/detector.go

```diff
@@ -0,0 +1,112 @@
+package ordersettler
+
+import (
+	"context"
+	"fmt"
+	"github.com/skip-mev/go-fast-solver/shared/contracts/fast_transfer_gateway"
+	"math/big"
+	"strconv"
+
+	"github.com/skip-mev/go-fast-solver/shared/clientmanager"
+	"github.com/skip-mev/go-fast-solver/shared/config"
+)
+
+type PendingSettlement struct {
+	SourceChainID      string
+	DestinationChainID string
+	OrderID            string
+	Amount             *big.Int
+	Profit             *big.Int
+}
+
+// DetectPendingSettlements scans all chains for pending settlements that need to be processed
+func DetectPendingSettlements(
+	ctx context.Context,
+	clientManager *clientmanager.ClientManager,
+	ordersSeen map[string]bool,
+) ([]PendingSettlement, error) {
+	var pendingSettlements []PendingSettlement
+
+	cosmosChains, err := config.GetConfigReader(ctx).GetAllChainConfigsOfType(config.ChainType_COSMOS)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Cosmos chains: %w", err)
+	}
+
+	var chains []config.ChainConfig
+	for _, chain := range cosmosChains {
+		if chain.FastTransferContractAddress != "" {
+			chains = append(chains, chain)
+		}
+	}
+
+	for _, chain := range chains {
+		bridgeClient, err := clientManager.GetClient(ctx, chain.ChainID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get client: %w", err)
+		}
+
+		fills, err := bridgeClient.OrderFillsByFiller(ctx, chain.FastTransferContractAddress, chain.SolverAddress)
+		if err != nil {
+			return nil, fmt.Errorf("getting order fills: %w", err)
+		}
+
+		for _, fill := range fills {
+			if ordersSeen != nil && ordersSeen[fill.OrderID] {
+				continue
+			}
+
+			sourceChainID, err := config.GetConfigReader(ctx).GetChainIDByHyperlaneDomain(strconv.Itoa(int(fill.SourceDomain)))
+			if err != nil {
+				continue
```

Comment:
the existing code has logic here that marks an order as seen. please go and double check these functions are logically equivalent pre and post refactor